boa/util/abi.py

```diff
@@ -1,15 +1,75 @@
 # wrapper module around whatever encoder we are using
-from typing import Any
+from typing import Annotated, Any
 
+from eth.codecs.abi import nodes
 from eth.codecs.abi.decoder import Decoder
 from eth.codecs.abi.encoder import Encoder
 from eth.codecs.abi.exceptions import ABIError
 from eth.codecs.abi.nodes import ABITypeNode
 from eth.codecs.abi.parser import Parser
+from eth_typing import Address as PYEVM_Address
+from eth_utils import to_canonical_address, to_checksum_address
+
+from boa.util.lrudict import lrudict
 
 _parsers: dict[str, ABITypeNode] = {}
 
 
+# XXX: inherit from bytes directly so that we can pass it to py-evm?
+# inherit from `str` so that ABI encoder / decoder can work without failing
```

Comment:
oh cool, we can remove the inheritance from `str` now (and maybe replace with inheritance from bytes)